### DISCUSSION TITLE
Docker: use command instead of entrypoint

### DISCUSSION THREAD

**Discussion by J0WI** [grafana/grafana#38361] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Move the `grafana-server` command to a [CMD](https://docs.docker.com/engine/reference/builder/#cmd). The entrypoint script should only be executed if grafana is started as server and execute the specified command (or the default CMD). E.g. do something like:

```bash
if [ "$1" = "grafana-server" ]; then
    # do stuff
fi

exec "$@"
```

**Why is this needed**:
The current version starts grafana in the entrypoint:
https://github.com/grafana/grafana/blob/f2c3d23140ca8eedb7d3881b1398d6888aaa1ef6/packaging/docker/run.sh#L80
This is bad because it does not allow to run the container with a custom command (`docker run  grafana/grafana whoami`).

**Comment by xlson** [comment]
I agree that using the endpoint is not ideal as it makes it hard to run the grafana image with bash or something else as the entrypoint. It's not impossible (`docker run -ti --entrypoint /bin/bash grafana/grafana`) but requires overriding the entrypoint manually which is harder than just throwing on a bash at the end of the command.

The problem with changing this behavior is that it would be a breaking change for users that rely on `run.sh` being the entrypoint. `run.sh` is written in such a way as any commands are sent to `grafana-server` when starting. I find it hard to say if the change is worth potentially breaking some users grafana setups.

I'll leave this issue open an see if we get more feedback on it. There's a case to be made for changing as well as keeping it the same.

**Comment by dnck** [comment]
I'm going to bump this feature-request. I'm trying to run grafana as container in a k8s pod witha  few other containers, and I have to do some nasty hacks to get grafana to start because of the entrypoint. I can elaborate if you need.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Reply by J0WI** [reply]
This is still an issue. Noting changed in the last years.
